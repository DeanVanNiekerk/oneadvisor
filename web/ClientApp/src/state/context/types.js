// @flow

export type Application = {
    id: string,
    name: string,
    color: string,
    relativePath: string,
    icon: string,
    isCurrent: boolean
}