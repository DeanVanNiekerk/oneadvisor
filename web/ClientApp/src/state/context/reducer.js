import { DIRECTORY_ID } from 'config/application'


const createApplication = (id, name, color, relativePath) => ({
  id,
  name,
  color,
  relativePath
})

export const defaultState = {
  applications: [
    createApplication(DIRECTORY_ID, "Directory", "#3949ab", "/directory")
  ]
}

export default (state = defaultState, action) => {

  switch (action.type) {
    default: return state;
  }

};