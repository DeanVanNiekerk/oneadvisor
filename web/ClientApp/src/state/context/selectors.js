import { createSelector } from 'reselect'
import { DEFAULT_APPLICATION_ID } from 'config/application'

const pathNameSelector = state => state.router.location.pathName
const appsSelector = state => state.context.applications

export const applicationsSelector = createSelector(
    pathNameSelector,
    appsSelector,
    (pathName, applications) => {

        return applications.map(app => {
            return {
                ...app,
                isCurrent: isCurrentApplication(app, pathName)
            }
        })

    }
)

const isCurrentApplication = (application, pathName) => {

    if(!pathName && application.id === DEFAULT_APPLICATION_ID)
        return true;

    return (pathName.indexOf(application.relativePath) !== -1);

}