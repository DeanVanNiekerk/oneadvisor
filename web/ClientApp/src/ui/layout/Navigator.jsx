import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
    Navbar,
    NavbarBrand,
    Nav,
    NavItem,
    NavLink
} from 'reactstrap';

const Navigator = ({ onLogout }) => (

    <Navbar color="primary" dark expand="md" sticky="top" >
        <NavbarBrand href="/">
            One Advisor
            </NavbarBrand>
        <Nav className="mr-auto" navbar>
            <NavItem>
                <Link className="nav-link" to="/directory/users">Users</Link>
            </NavItem>
            <NavItem>
                <NavLink href="#" onClick={onLogout}>Signout</NavLink>
            </NavItem>
        </Nav>
    </Navbar>
);

Navigator.propTypes = {
    onLogout: PropTypes.func.isRequired
};

export default Navigator;



