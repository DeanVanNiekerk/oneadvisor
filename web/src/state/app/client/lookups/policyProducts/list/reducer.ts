import { PolicyProduct } from '../types';
import { PolicyProductListAction } from './actions';

export type State = {
    readonly items: PolicyProduct[];
    readonly fetching: boolean;
};

export const defaultState: State = {
    items: [],
    fetching: false,
};

export const reducer = (
    state: State = defaultState,
    action: PolicyProductListAction
): State => {
    switch (action.type) {
        case "POLICYPRODUCTS_LIST_RECEIVE": {
            return {
                ...state,
                items: action.payload,
                fetching: false,
            };
        }
        case "POLICYPRODUCTS_LIST_FETCHING": {
            return {
                ...state,
                fetching: true,
            };
        }
        case "POLICYPRODUCTS_LIST_FETCHING_ERROR": {
            return {
                ...state,
                items: [],
                fetching: false,
            };
        }
        default:
            return state;
    }
};
