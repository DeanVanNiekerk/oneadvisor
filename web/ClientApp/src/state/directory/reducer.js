import { combineReducers } from 'redux';

import { reducer as users } from './users/reducer';

export default combineReducers({
  users,
});