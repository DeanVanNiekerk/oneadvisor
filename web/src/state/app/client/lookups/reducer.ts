import { combineReducers } from 'redux';

import { reducer as all, State as LookupsState } from './all/reducer';
import { reducer as clientTypes, State as ClientTypesState } from './clientTypes/reducer';
import { reducer as contactTypes, State as ContactTypesState } from './contactTypes/reducer';
import { reducer as marritalStatus, State as MarritalStatusState } from './marritalStatus/reducer';
import { reducer as policyProductTypes, State as PolicyProductTypesState } from './policyProductTypes/reducer';
import { reducer as policyTypes, State as PolicyTypesState } from './policyTypes/reducer';

export type State = {
    all: LookupsState;
    contactTypes: ContactTypesState;
    marritalStatus: MarritalStatusState;
    policyTypes: PolicyTypesState;
    clientTypes: ClientTypesState;
    policyProductTypes: PolicyProductTypesState;
};

export const reducer = combineReducers({
    all: all,
    marritalStatus: marritalStatus,
    contactTypes: contactTypes,
    policyTypes: policyTypes,
    clientTypes: clientTypes,
    policyProductTypes: policyProductTypes,
});
