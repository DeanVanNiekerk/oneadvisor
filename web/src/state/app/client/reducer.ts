import { combineReducers } from 'redux';

import { reducer as clients, State as ClientsState } from './clients/reducer';
import { reducer as contacts, State as ContactsState } from './contacts/reducer';
import { reducer as clientImport, State as ClientImportState } from './import/reducer';
import { reducer as lookups, State as LookupsState } from './lookups/reducer';
import { reducer as policies, State as PoliciesState } from './policies/reducer';

export type State = {
    clients: ClientsState;
    import: ClientImportState;
    policies: PoliciesState;
    contacts: ContactsState;
    lookups: LookupsState;
};

export const reducer = combineReducers({
    clients: clients,
    import: clientImport,
    policies: policies,
    contacts: contacts,
    lookups: lookups,
});
