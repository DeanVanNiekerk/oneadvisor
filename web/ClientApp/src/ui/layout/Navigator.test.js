import React from 'react';
import Navigator from './Navigator';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom'
import configureMockStore from 'redux-mock-store'
import { defaultState } from 'state/context/reducer'

const mockStore = configureMockStore()

describe('Navigator', () => {

    it('renders correctly', () => {

        const store = mockStore({
            context: {
                ...defaultState
            },
            router: {
                location: {
                    pathName: ""
                }
            }
        })

        const tree = renderer
            .create(<Router><Navigator store={store} auth={{}} onLogout={() => {}} /></Router>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    // it('navvar toggles on click', () => {

    //     const wrapper = mount(<Router><Navigator auth={{}} /></Router>).find(Navigator);
    //     const instance = wrapper.instance();

    //     const navbarTogger = wrapper.find(".navbar-toggler");

    //     //nav is initially closed
    //     expect(instance.state.isOpen).toBe(false);

    //     navbarTogger.simulate('click');

    //     //nav should now be open
    //     expect(instance.state.isOpen).toBe(true);
    // });

});