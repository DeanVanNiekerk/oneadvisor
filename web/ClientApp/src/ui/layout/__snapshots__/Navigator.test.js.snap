// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator renders correctly 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-primary sticky-top"
>
  <a
    className="navbar-brand"
    href="/"
  >
    One Advisor
  </a>
  <ul
    className="mr-auto navbar-nav"
  >
    <li
      className="nav-item sc-bdVaJa bWkJfu"
    >
      <a
        className="nav-link active"
        href="/directory"
        onClick={[Function]}
      >
        Directory
      </a>
    </li>
  </ul>
  <ul
    className="ml-auto navbar-nav"
  >
    <li
      className="sc-bdVaJa bWkJfu"
    >
      <a
        className="nav-link"
        href="#"
        onClick={[Function]}
      >
        Signout
      </a>
    </li>
  </ul>
</nav>
`;
